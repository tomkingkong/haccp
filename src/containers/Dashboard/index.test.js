import React from 'react';
import { shallow } from 'enzyme';

import { Dashboard } from '.';

describe('Dashboard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Dashboard />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default userProducts state', () => {
    const expected = [];
    expect(wrapper.state().userProducts).toEqual(expected);
  });

  it('should userProducts state on mount', () => {
    const mockData = [
      { name: 'Meat Sticks', id: 1 },
      { name: 'Canned Ham', id: 2 },
      { name: 'SnickleFritz', id: 3 }
    ];
    wrapper = shallow(<Dashboard products={mockData} />);
    wrapper.instance().componentDidMount();
    expect(wrapper.state().userProducts).toEqual(mockData);
  });
});