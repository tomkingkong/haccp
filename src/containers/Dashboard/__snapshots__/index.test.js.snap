// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match snapshot 1`] = `
<div>
  <h2>
    Dashboard
  </h2>
  <div>
    <h4
      className="products-title"
    >
      Products
    </h4>
    <Connect(ProductContainer)
      userProducts={Array []}
    />
  </div>
</div>
`;
